subsys/power/power.c

```diff
@@ -198,6 +204,58 @@ void pm_system_resume(void)
 	}
 }
 
+ssize_t pm_constraint_set(enum pm_state state)
+{
+	ssize_t ret;
+
+	if (state < 0 || state > PM_STATES_LEN) {
+		return -EINVAL;
+	}
+
+	k_mutex_lock(&pm_constraint_lock, K_FOREVER);
+
+	if (size_add_overflow(pm_constraints[state], 1,
```

Comment:
I'm really not sure this is worth doing.  Yes, it's technically possible for somebody to set the same constraint 2^31 times, and this would prevent that from introducing undefined behavior.

However, it also means that we should have an expectation that people will not on check results for errors but be able to react in some reasonable way if they occur.